client/components/domains/map-domain-step/index.jsx

```diff
@@ -31,39 +33,39 @@ const MapDomainStep = React.createClass( {
 		domainsWithPlansOnly: React.PropTypes.bool.isRequired,
 		onRegisterDomain: React.PropTypes.func.isRequired,
 		onMapDomain: React.PropTypes.func.isRequired
-	},
+	};
 
-	getInitialState: function() {
-		return { searchQuery: this.props.initialQuery };
-	},
+	state = {
+		searchQuery: this.props.initialQuery
+	};
 
-	componentWillMount: function() {
+	componentWillMount() {
 		if ( this.props.initialState ) {
 			this.setState( Object.assign( {}, this.props.initialState, this.getInitialState() ) );
```

Comment:
This fails if you try to go back from the plans step to the themes one (for some reason it skips over the domains step...) and then visit the domains step again - because `this.getInitialState()` no longer exists :)